07/reshape-a-matrix.md

```diff
@@ -2,137 +2,218 @@
 
 ##### !end-question
+##### !hint
 
-##### !placeholder
+Further questions to ask as you read through the problem statement and examples:
+- What is the goal of the function?
+- What are the types of the expected inputs and outputs?
+- Are there any restrictions on any of the inputs?
+  - For example: if any of the inputs are a list, do we know anything about how the list is ordered?
+- What do the examples show us about the data types and values that are allowed for our inputs?
+- What do the examples tell us about the return value in different scenarios?
+- Reflecting on the observations you have made so far, what questions would give you new information?
 
-Give three clarifying questions.
+<br>
 
-##### !end-placeholder
+Consider the following for inspiration:
+- [About PSEs](../about-pses/about-pses.md)
+- [Our example PSE with example answers](../about-pses/example-pse.md)
+- Previous PSEs
 
-<!-- other optional sections -->
-<!-- !hint - !end-hint (markdown, users can see after a failed attempt) -->
-<!-- !rubric - !end-rubric (markdown, instructors can see while scoring a checkpoint) -->
+##### !end-hint
 ##### !explanation
 
-Here are some example clarifying questions:
+One of many possible responses could look like:
+
+1. The last two bullet points of the description say that we should output a new matrix or raise a ValueError.
+    - This tells me that the only expected return value of the function should be a new matrix if reshaping is possible.
 
-1.  Are there any constraints on matrix sizes?
-1.  Can a matrix have 1 row or 1 column?
-1.  What if the matrix is empty or `None`?
+2. The problem statement says the matrix we receive as an argument will be represented as a 2 dimensional array. In the examples, the outputs are also displayed as 2 dimensional arrays.
+    - This means that the new matrix we create should also be a 2 dimensional array.
+
+3. When discussing the output, it says we should raise a ValueError if the original matrix cannot be reshaped as requested.
+    - I will need to do some check in my function to see if the number of items in the original matrix is equal to the number of items in the requested matrix before trying the reshape operation.
+
+4. The problem statement says the new matrix needs to be filled with the elements of the old matrix in the same row-traversing order.
+    - This means that the order of the elements in the new matrix matters. I will need to go row by row starting at the front of each list in the original matrix as I fill up the new matrix.
+
+5. We can assume that the arguments to the function will be valid, so the input matrix will be a 2 dimensional list, and the `rows` and `columns` will not be negative. We have also been given the note "It can be assumed that all given `rows` and `columns` are positive numbers." 
```

Comment:
2 dimensional list just doesn't sit well with me. Like I'd be totally OK with 2 dimensional array, even though there would be no structural difference. In the grand scheme of things, this isn't a big deal, but here's an alternative phrasing to consider, which also helps the reader rule out jagged inputs.
```suggestion
5. We can assume that the arguments to the function will be valid, so the input matrix will be a nested list with a consistent number of `rows` and `columns`, and the `rows` and `columns` will not be negative. We have also been given the note "It can be assumed that all given `rows` and `columns` are positive numbers." 
```